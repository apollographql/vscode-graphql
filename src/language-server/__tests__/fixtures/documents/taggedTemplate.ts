// hint for reading the snapshots generated by this file:
// it is important to check that a rule doesn't only enter `meta.embedded.block.graphql`
// but also leaves it again

declare function gql<A = any, B = any>(strings: TemplateStringsArray): void;
declare function foo<A = any, B = any>(strings: TemplateStringsArray): void;
declare type SomeResult = any;
declare type SomeVariables = any;

// for comparison - this is what a normal function all is colored
foo`query Foo { test }`;
foo<SomeResult, SomeVariables>`query Foo { test }`;

// prettier-ignore
gql`query Q1 { test }`

// prettier-ignore
gql<SomeResult, SomeVariables>`query Q2 { test }`

// prettier-ignore
gql  <SomeResult, SomeVariables>  `
query Q3 { test }`

// prettier-ignore
gql    `query Q4 { test }`

gql`
  query Q5 {
    test
  }
`;

gql<SomeResult, SomeVariables>`
  query Q6 {
    test
  }
`;

// prettier-ignore
gql   `
  query Q7 {
    test
  }
`;

// syntax highlighting cannot work in all examples after this - textmate grammars don't work over multiple lines like this

// prettier-ignore
gql
`query Q8 { test }`

// prettier-ignore
gql<{
  test: string;
},{
  test: string;
}>`query Q9 { test }`

gql<
  {
    test: string;
  },
  {
    test: string;
  }
>`
  query Q10 {
    test
  }
`;

// prettier-ignore
gql
`
  query Q11 {
    test
  }
`;

export {};
