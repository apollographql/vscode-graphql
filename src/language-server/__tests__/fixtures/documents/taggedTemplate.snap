>declare function gql<A = any, B = any>(strings: TemplateStringsArray): void;
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql
>declare type SomeResult = any;
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql
>declare type SomeVariables = any;
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql`query { test }`
#^^^ inline.graphql apollo.taggedTemplate entity.name.function.tagged-template.js
#   ^ inline.graphql apollo.taggedTemplate
#    ^^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql keyword.operation.graphql
#         ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#          ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
#           ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#            ^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
#                ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#                 ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
#                  ^ inline.graphql apollo.taggedTemplate punctuation.definition.string.template.end.js
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql<SomeResult, SomeVariables>`query { test }`
#^^^ inline.graphql apollo.taggedTemplate entity.name.function.tagged-template.js
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql apollo.taggedTemplate
#                               ^^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql keyword.operation.graphql
#                                    ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#                                     ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
#                                      ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#                                       ^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
#                                           ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#                                            ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
#                                             ^ inline.graphql apollo.taggedTemplate punctuation.definition.string.template.end.js
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql<{
#^^^^^^ inline.graphql
>  test: string;
#^^^^^^^^^^^^^^^^ inline.graphql
>},{
#^^^^ inline.graphql
>  test: string;
#^^^^^^^^^^^^^^^^ inline.graphql
>}>`query { test }`
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql
#^^^^ inline.graphql
>`query { test }`
#^^^^^^^^^^^^^^^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql    `query { test }`
#^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql
>
>gql`
#^^^ inline.graphql apollo.taggedTemplate entity.name.function.tagged-template.js
#   ^ inline.graphql apollo.taggedTemplate
>  query {
#^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql
#  ^^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql keyword.operation.graphql
#       ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#        ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
>    test
#^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#    ^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
>  }
#^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#  ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
>`;
#^ inline.graphql apollo.taggedTemplate punctuation.definition.string.template.end.js
# ^^ inline.graphql
>
>gql<SomeResult, SomeVariables>`
#^^^ inline.graphql apollo.taggedTemplate entity.name.function.tagged-template.js
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql apollo.taggedTemplate
>  query {
#^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql
#  ^^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql keyword.operation.graphql
#       ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#        ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
>    test
#^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#    ^^^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
>  }
#^^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql
#  ^ inline.graphql apollo.taggedTemplate meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
>`;
#^ inline.graphql apollo.taggedTemplate punctuation.definition.string.template.end.js
# ^^ inline.graphql
>
>gql<
#^^^^^ inline.graphql
>  {
#^^^^ inline.graphql
>    test: string;
#^^^^^^^^^^^^^^^^^^ inline.graphql
>  },
#^^^^^ inline.graphql
>  {
#^^^^ inline.graphql
>    test: string;
#^^^^^^^^^^^^^^^^^^ inline.graphql
>  }
#^^^^ inline.graphql
>>`
#^^^ inline.graphql
>  query {
#^^^^^^^^^^ inline.graphql
>    test
#^^^^^^^^^ inline.graphql
>  }
#^^^^ inline.graphql
>`;
#^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql
#^^^^ inline.graphql
>`
#^^ inline.graphql
>  query {
#^^^^^^^^^^ inline.graphql
>    test
#^^^^^^^^^ inline.graphql
>  }
#^^^^ inline.graphql
>`;
#^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql   `
#^^^^^^^^ inline.graphql
>  query {
#^^^^^^^^^^ inline.graphql
>    test
#^^^^^^^^^ inline.graphql
>  }
#^^^^ inline.graphql
>`;
#^^^ inline.graphql
>